src/Features/Core/Portable/ChangeSignature/AbstractChangeSignatureService.cs

```diff
@@ -126,15 +126,9 @@ public async Task<ImmutableArray<ChangeSignatureCodeAction>> GetChangeSignatureC
                 }
                 else if (typeSymbol.IsRecord)
                 {
-                    // A bit hacky to determine the parameters of primary constructor associated with a given record.
-                    // TODO: record structs.
-                    const int RecordDeclarationRawKind = 9063;
-                    var potentialPrimaryCtor = typeSymbol.InstanceConstructors.FirstOrDefault(
-                        c => c.DeclaringSyntaxReferences.FirstOrDefault()?.GetSyntax()?.RawKind == RecordDeclarationRawKind);
-
-                    if (potentialPrimaryCtor is not null)
+                    if (TryGetRecordPrimaryConstructor(typeSymbol, out var primaryConstructor))
                     {
-                        symbol = potentialPrimaryCtor;
+                        symbol = primaryConstructor;
```

Comment:
@CyrusNajmabadi I forgot to add `[NotNullWhen(true)]` to the out parameter, but... There is NO warning here, which is weird. `symbol` is here assigned a "maybe null" value, and is being dereferenced a few lines later.